### ISSUE TITLE
No to overwrite authorized_keys file on the distant host but still get key Management Enabled

### ISSUE THREAD

**Issue by kumaip** [skavanagh/KeyBox#200]
Hello,
First of all, thanks a lot for this solution which is very useful for us.
I wanna know if it is possible for keybox not to overwrite the authorized_keys file (but only add a new entry) on the distant host but still have the key management function enabled?  
Thanks in advance for your response.
Regards
Y

**Comment by skavanagh** [comment]
No, it either manages the keys or doesn't. There isn't an in-between that just does provisioning. It's of course possible to add what's in the authorized keys file to keybox, before you register the system.

**Comment by kumaip** [comment]
Ok, thanks for your reply. 
I get another question. Now I wanna connect to the database to reset admin password, but when I use H2 console to connect to  "jdbc:h2:tcp://xxxx:8082/~/keybox" I get connection refused message.
Is this the right way to connect to the keybox database?

**Comment by skavanagh** [comment]
Well that depends on what it's set to in the KeyBoxConfig.properties
Assuming you left the default it should be
```
jdbc:h2:/<path>/KeyBox-jetty/jetty/keybox/WEB-INF/classes/keydb/keybox;CIPHER=AES;
```
the user is 'keybox' and when you set the password it's 'filepwd \<password\>'

**Comment by kumaip** [comment]
![screenshot - 01262017 - 09 47 34 am](https://cloud.githubusercontent.com/assets/25325788/22325010/84ebc33a-e3ac-11e6-8b02-65c4a8026903.png)
Thanks a lot. I managed to connect to the database (I think maybe) but I can't find where to reset the admin password

**Comment by skavanagh** [comment]
Passwords are stored with SHA-256 with a salt.
```
update users set password='BXugPWxEEEhj3HNh/kV4ll0YhzYPkKCJWILlimJI/IY=', salt=null, otp_secret=null where username='admin';
```
should set back to 'changeme'

**Comment by skavanagh** [comment]
As far as connect to the correct database, you need to make sure the path is correct.  
```
jdbc:h2:/<path>/KeyBox-jetty/jetty/keybox/WEB-INF/classes/keydb/keybox
```
It should be pointed to the keybox database in the  KeyBox-jetty/jetty/keybox/WEB-INF/classes/keydb folder.

Where is it creating the DB in relation to the keydb folder??

**Comment by nickmisasi13** [comment]
@kumaip https://github.com/nickmisasi13/KeyBox I made some (hacky) tweaks to Sean's code that allows idempotent SSH key management (It won't overwrite existing keys in the target authorized_keys file), it's only on version 2.86 but I imagine if you look through the commits the same changes could be made to the latest version from this repo
